nixos/modules/programs/rush.nix

```
@@ -0,0 +1,127 @@
+        statements located between it and the next rule statement (or end of file, whichever occurs first)
+        modify the definition of that rule.
+      '';
+    };
+
+    shell = lib.mkOption {
+      readOnly = true;
+      type = either shellPackage path;
+
+      description = ''
+        The resolved shell path that users can inherit to set `rush` as their login shell.
+        This is a convenience option for use in user definitions. Example:
+          `users.users.alice = { inherit (config.programs.rush) shell; ... };`
+      '';
+    };
+
+    wrap = lib.mkOption {
+      type = bool;
+      default = config.security.enableWrappers;
+      defaultText = lib.literalExpression "config.security.enableWrappers";
+
+      description = ''
+        Whether to wrap the `rush` binary with a SUID-enabled wrapper.
+        This is required if {option}`security.enableWrappers` is enabled in your configuration.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      (lib.mkIf (!lib.versionAtLeast cfg.package.version "2.0") {
+        warnings = lib.optional (!builtins.hasAttr "rush.rc" config.environment.etc) ''
+          The selected version of GNU Rush (${cfg.package.version}) is older than 2.0 and does not support configuration via Nix.
+
+          A legacy `/etc/rush.rc` file is required for versions < 2.0, but this file will **not** be created automatically by this module.
+
+          To remove this warning:
+            - Either upgrade to GNU Rush 2.0 or newer, **or**
+            - Manually provide a valid `/etc/rush.rc` file via `environment.etc."rush.rc"` in your configuration.
+        '';
+      })
+
+      (lib.mkIf (lib.versionAtLeast cfg.package.version "2.0") {
+        environment.etc."rush.rc".text =
+          lib.pipe
+            [
+              "# This file was created by the module `programs.rush`;"
+              "rush 2.0"
+              (lib.optionalString (opt.global.isDefined && cfg.global != "") "global\n${indent cfg.global}")
+              (lib.optionals (opt.rules.isDefined && cfg.rules != { }) (
+                lib.mapAttrsToList (name: content: "rule ${name}\n${indent content}") cfg.rules
+              ))
+            ]
+            [
+              (lib.flatten)
+              (builtins.filter (line: line != ""))
+              (builtins.concatStringsSep "\n\n")
+              (lib.mkDefault)
+            ];
+      })
+
+      (lib.mkIf cfg.wrap {
+        security.wrappers.rush = lib.mkDefault {
+          group = "root";
+          owner = "root";
+          permissions = "u+rx,g+x,o+x";
+          setgid = false;
+          setuid = true;
+          source = lib.getExe cfg.package;
+        };
+      })
+
+      rec {
+        programs.rush.shell = if cfg.wrap then config.security.wrapperDir + "/rush" else cfg.package;
+
+        environment = {
+          shells = [ programs.rush.shell ];
```

```suggestion
      {
        programs.rush.shell = if cfg.wrap then config.security.wrapperDir + "/rush" else cfg.package;

        environment = {
          shells = [ cfg.shell ];
```
The rec is unnecessary, since you already use `cfg` everywhere else.